Dockerfile

```diff
@@ -2,31 +2,41 @@
 ARG base_image=ruby:2.7.6-slim-buster
 
 FROM $base_image AS builder
+
 # TODO: have a separate build image which already contains the build-only deps.
-RUN apt-get update -qy && apt-get upgrade -y
-# TODO: Look to remove Node v9 (below) (E2E tests currently fail without it).
-RUN curl -sL https://deb.nodesource.com/setup_9.x | bash -
-RUN apt-get update && apt-get install -y build-essential nodejs git npm && \
-    npm install -g phantomjs-prebuilt@2 --unsafe-perm
+RUN apt-get update -qy && apt-get upgrade -y && apt-get install -y build-essential curl git
+
+RUN curl -sL https://deb.nodesource.com/setup_lts.x | bash -
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+
+RUN apt-get update && apt-get install -y nodejs yarn && apt-get clean
+
 # TODO: Can ASSETS_PREFIX default to `/assets/publisher` within Publisher?
 ENV RAILS_ENV=production GOVUK_APP_NAME=publisher ASSETS_PREFIX=/assets/publisher
 RUN mkdir /app
+
 WORKDIR /app
-COPY Gemfile Gemfile.lock .ruby-version /app/
+COPY Gemfile Gemfile.lock .ruby-version package.json yarn.lock /app/
+
 RUN bundle config set deployment 'true'
 RUN bundle config set without 'development test'
 RUN bundle install -j8 --retry=2
+RUN yarn install --production --frozen-lockfile
 COPY . /app
+
 # TODO: We probably don't want assets in the image; remove this once we have a proper deployment process which uploads to (e.g.) S3.
 RUN GOVUK_APP_DOMAIN=www.gov.uk GOVUK_WEBSITE_ROOT=www.gov.uk bundle exec rails assets:precompile
 
 FROM $base_image
-RUN apt-get update -qy && apt-get upgrade -y && \
-    apt-get install -y nodejs
+
 # TODO: MONGODB_URI shouldn't be set here but seems to be required by E2E tests, figure out why.
 # TODO: Can ASSETS_PREFIX default to `/assets/publisher` within Publisher?
 ENV GOVUK_PROMETHEUS_EXPORTER=true RAILS_ENV=production GOVUK_APP_NAME=publisher ASSETS_PREFIX=/assets/publisher MONGODB_URI=mongodb://mongo/govuk_content_development
+
+COPY --from=builder /usr/bin/node* /usr/bin/
```

Comment:
@kevindew does the runner image _need_ node installed? Just wondering why we don't also need yarn, in that case. Sprockets assets are compiled within the builder image and copied across – so I suspect node might be unnecessary. 🤔

Reply:
Good shout. I'd like to think it doesn't, but often it seems apps have a surprising need for a node runtime. I reckon, with the e2e tests, that this image is tested well enough that I can try remove it and we'll find out if it's a problem.

Reply:
Looks like it's not happy 

<img width="948" alt="image" src="https://user-images.githubusercontent.com/42515961/172579121-81588439-21b5-4acc-a513-9a307b9fb50f.png">

Reply:
Yep – it looks like the `uglifier` gem, which is included in production, depends on `execjs` gem, which in turn depends on Node.js 😔

<img width="1161" alt="ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime." src="https://user-images.githubusercontent.com/7735945/172580608-decc2843-0c19-4b38-8d38-d0c69c4cc808.png">

As for whether `uglifier` is really needed in production... I don't know? But since it's already there, I guess the simplest thing is just to give the runtime image a copy of Node.js

Reply:
Thanks for digging out that detail. 

That is a bit annoying. You do need uglifier for production, so that you can run assets:precompile in production - however it seems a bit frustrating that it demands it outside of runtime usage.